sdks/java/extensions/sketching/src/main/java/org/apache/beam/sdk/extensions/sketching/TDigestQuantiles.java

```diff
@@ -0,0 +1,321 @@
+ * build  {@code PCollection} of {@link KV}s with each pair corresponding to
+ * a couple (quantile, value).
+ *
+ * <pre><code>
+ * {@literal PCollection<MergingDigest>} pc = ...;
+ * {@literal PCollection<KV<Double, Double>>} quantiles = pc.apply(ParDo.of(
+ *        {@literal new DoFn<MergingDigest, KV<Double, Double>>()} {
+ *          {@literal @ProcessElement}
+ *           public void procesElement(ProcessContext c) {
+ *             Double[] quantiles = {0.01, 0.25, 0.5, 0.75, 0.99}
+ *             for (Double q : quantiles) {
+ *                c.output(KV.of(q, c.element().quantile(q));
+ *             }
+ *           }}));
+ * </code></pre>
+ *
+ * <p>One can also retrieve the approximate quantile position of a given element in the stream
+ * using {@code cdf(double)} method instead of {@code quantile(double)}.
+ *
+ * <h3>Example 4: Using the CombineFn</h3>
+ *
+ * <p>The {@code CombineFn} does the same thing as the {@code PTransform}s but
+ * it can be used for doing stateful processing or in
+ * {@link org.apache.beam.sdk.transforms.CombineFns.ComposedCombineFn}.
+ *
+ * <p>This example is not really interesting but it shows how one can properly
+ * create a {@link TDigestQuantilesFn}.
+ *
+ * <pre><code>
+ *  double cf = 250;
+ * {@literal PCollection<Double>} input = ...;
+ * {@literal PCollection<MergingDigest>} output = input.apply(Combine
+ *         .globally(TDigestQuantilesFn.create(cf)));
+ * </code></pre>
+ *
+ * <p><b>Warning: this class is experimental.</b> <br>
+ * Its API is subject to change in future versions of Beam.
+ * */
+@Experimental
+public final class TDigestQuantiles {
+
+  /**
+   * Compute the stream in order to build a T-Digest structure (MergingDigest)
+   * for keeping track of the stream distribution and returns a {@code PCollection<MergingDigest>}.
+   * <br> The resulting structure can be queried in order to retrieve the approximate value
+   * at a given quantile or the approximate quantile position of a given element.
+   */
+  public static GlobalDigest globally() {
+    return GlobalDigest.builder().build();
+  }
+
+  /**
+   * Like {@link #globally()}, but builds a digest for each key in the stream.
+   *
+   * @param <K>             the type of the keys
+   */
+  public static <K> PerKeyDigest<K> perKey() {
+    return PerKeyDigest.<K>builder().build();
+  }
+
+  /** Implementation of {@link #globally()}. */
+  @AutoValue
+  public abstract static class GlobalDigest
+          extends PTransform<PCollection<Double>, PCollection<MergingDigest>> {
+
+    abstract double compression();
+
+    abstract Builder toBuilder();
+
+    static Builder builder() {
+      return new AutoValue_TDigestQuantiles_GlobalDigest.Builder()
+              .setCompression(100);
+    }
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      abstract Builder setCompression(double cf);
+
+      abstract GlobalDigest build();
+    }
+
+    public GlobalDigest withCompression(double cf) {
```

Comment:
Please add a javadoc here.

Reply:
ok, I also added javadoc in the other classes.